src/components/ServerList.js

```diff
@@ -0,0 +1,121 @@
+import React, { Component } from 'react';
+import { Button, Icon, Spinner } from 'network-canvas-ui';
+import ServerDiscoverer from '../utils/serverDiscoverer';
+import ServerCard from '../components/ServerCard';
+
+class ServerList extends Component {
+  constructor() {
+    super();
+
+    this.state = {
+      error: null,
+      servers: [],
+    };
+  }
+  componentDidMount() {
+    window.addEventListener('online', this.handleNetworkChange);
+    window.addEventListener('offline', this.handleNetworkChange);
+
+    this.initServer();
+  }
+
+  componentDidUpdate() {
+    // Give some console output about the error, just for debugging.
+    if (this.state.error) {
+      // eslint-disable-next-line
+      console.warn(this.state.error);
+    }
+  }
+
+  componentWillUnmount() {
+    window.removeEventListener('online', this.handleNetworkChange);
+    window.removeEventListener('offline', this.handleNetworkChange);
+    this.unbindServerEvents();
+  }
+
+  bindServerEvents = () => {
+    if (!this.serverDiscoverer) { return; }
+
+    this.serverDiscoverer.on('SERVER_ANNOUNCED', (response) => {
+      this.setState(prevState => ({
+        servers: [...prevState.servers, response],
+      }));
+    });
+
+    this.serverDiscoverer.on('SERVER_REMOVED', (response) => {
+      console.log('Looking for:');
+      console.log(response);
+
+      this.setState(prevState => ({
+        // eslint-disable-next-line
+        servers: prevState.servers.filter(item => !(item.name == response.name && item.index == response.index)),
```

Comment:
I'm testing on iOS. When a service is removed, its index comes through as `-1`, so this never matches.

Reply:
Thanks Bryan - I've caught and fixed this in a later commit which I'll push up tomorrow.  React is a cruel mistress.